packages/next-env/index.ts

```diff
@@ -80,6 +80,8 @@ export function processEnv(
           typeof parsed[key] === 'undefined' &&
           typeof origEnv[key] === 'undefined'
         ) {
+          // We're being imprecise in the type system - assume parsed[key] can be undefined
+          // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
```

Comment:
See https://twitter.com/mattpocockuk/status/1681267079977000961 from our friend @mattpocock about why I'm not going to try to get this working perfectly 😄

Reply:
How about changing `Object.keys()` to `Object.entries()`?

Reply:
That might work! I've learned the hard way to not try to change _runtime_ behavior in _static analysis_ oriented PRs like this one. If you really want me to make this change I can - but would suggest moving it to a separate PR as a separate cleanup item.